solana/solana-ibc/programs/solana-ibc/src/storage.rs

```diff
@@ -431,7 +452,7 @@ macro_rules! from_ctx {
             private,
             provable,
             chain,
-            accounts,
+            accounts: &[],
         })
     }}
 }
```

Comment:
What you wanna do here is add other invocations of the macro.
Something like:

```rust
macro_rules! from_ctx {
    ($ctx:expr) => {
        from_ctx!($ctx, Default::default())
    };
    ($ctx:expr with accounts) => {
        let account = {
            // put body of to_token_accounts here
        };
        from_ctx!($ctx, accounts)
    };

    ($ctx:expr, $accounts:expr) => {{
        let provable = storage::get_provable_from(&$ctx.accounts.trie)?;
        let chain = &mut $ctx.accounts.chain;

        // Before anything else, try generating a new guest block.  However, if
        // that fails it’s not an error condition.  We do this at the beginning
        // of any request.
        chain.maybe_generate_block(&provable)?;

        storage::IbcStorage::new(storage::IbcStorageInner {
            private: &mut $ctx.accounts.storage,
            provable,
            chain,
            accounts: $accounts,
        })
    }}
}
```

and then you should be able to do:

```
        let mut store = storage::from_ctx!(ctx with accounts);
```

Reply:
nice, makes sense. Could do that.